controllers/release/release_adapter.go

```diff
@@ -176,10 +212,38 @@ func (a *Adapter) EnsureTargetContextIsSet() (results.OperationResult, error) {
 			a.logger.Info("Running in a KCP environment")
 			a.targetContext = logicalcluster.WithCluster(a.context, logicalcluster.New(releasePlan.Spec.Target.Workspace))
 		}
+
+		a.syncer.SetContext(a.targetContext)
 	}
 	return results.ContinueProcessing()
 }
 
+// createOrUpdateSnapshotEnvironmentBinding creates or updates a SnapshotEnvironmentBinding for the Release being
+// processed.
+func (a *Adapter) createOrUpdateSnapshotEnvironmentBinding(releasePlanAdmission *v1alpha1.ReleasePlanAdmission) (*appstudioshared.ApplicationSnapshotEnvironmentBinding, error) {
+	components, snapshot, environment, err := a.getSnapshotEnvironmentResources(releasePlanAdmission)
+	if err != nil {
+		return nil, err
+	}
+
+	// The binding information would need to be updated no matter if it already exists or not
+	binding := gitops.NewSnapshotEnvironmentBinding(components, snapshot, environment)
+
+	// Search for an existing binding
+	existingBinding, err := a.getSnapshotEnvironmentBinding(environment, releasePlanAdmission)
+	if err != nil && !errors.IsNotFound(err) {
+		return nil, err
+	}
+
+	if existingBinding != nil {
+		patch := client.MergeFrom(existingBinding.DeepCopy())
+		existingBinding.Spec = binding.Spec
+		return existingBinding, a.client.Patch(a.targetContext, existingBinding, patch)
+	} else {
+		return binding, a.client.Create(a.targetContext, binding)
```

Comment:
we create the binding in the targetContext but on line 467 we search for it in the context